includes/Modules/Analytics.php

```diff
@@ -879,11 +897,18 @@ function ( $name ) {
 					return $request;
 				}
 
-				$date_ranges = array(
-					$this->parse_date_range(
+				$date_range = $data['dateRange'] ?: 'last-28-days';
+				$start_date = $data['startDate'];
+				$end_date   = $data['endDate'];
+				if ( ! strtotime( $start_date ) || ! strtotime( $end_date ) ) {
+					list( $start_date, $end_date ) = $this->parse_date_range(
 						$date_range,
 						$data['compareDateRanges'] ? 2 : 1
-					),
+					);
+				}
```

Comment:
I think this code would be a bit more clear to read if we just checked first `strtotime( $data['startDate'] ) && strtotime( $data['endDate'] )` and then used an `else` for the date range logic.

Reply:
I am not sure that I get what you want. If I do it, we will have repeated code which is not what we need, isn't it? Or I miss something?

Reply:
Oh I guess what I was meaning to say is: Let's only access `dateRange` in the if clause here. That should be the only place where it's needed, and it makes it more obvious when looking at the code that it is just a fallback.

Reply:
It's also needed when `multiDateRange` is set. See here: https://github.com/google/site-kit-wp/blob/c52626a6b6346c1a92496f552e3b9895c6d030b6/includes/Modules/Analytics.php#L926-L928

So, we can't move it into the if clause.

Reply:
@eugene-manuilov I see, however `multiDateRange` is also a legacy argument that we should only support if `dateRange` is provided. So this should go into the clause for when `startDate` and `endDate` do not exist.

This means that we'll at some point need to introduce further arguments for the secondary date range to make this available in combination with the new `startDate` and `endDate` arguments. Maybe `compareStartDate` and `compareEndDate`? Anyway, let's not do this now since it needs more definition.

Can you move the `multiDateRange` bit accordingly? With that, this should be good to go.

Reply:
Yep, updated. I think it should be good now.